doc/compiler/jitserver/Build.md

```diff
@@ -0,0 +1,144 @@
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+
+There are currently 3 supported build procedures. Select the one that best suits your needs.
+
+- [Building the whole VM](#vm). The complete build process (Most correct and recommended). Produces a complete JVM as output (including the JIT).
+- [Building only the JIT](#jit). Choose this if you have an existing SDK and have no need to modify the VM or JCL. This only builds OMR and everything in the `runtime/compiler` directory. Output artifact is `libj9jit29.so`. (Fastest way to get JITServer built)
+- [Building in Docker](#docker). Choose this if you don't want to deal with build dependencies. Takes the longest.
+
+For building the whole VM and building only the JIT, you will have to go through the [Prepare your system](#prerequisites) step.
+For building in Docker, you can skip [Prepare your system](#prerequisites).
+
+If you run into issues during this process, please document any solutions here.
+
+## PREREQUISITES
+- The following example is for Ubuntu18:
+
+```
+apt-get update \
+ && apt-get install -qq -y --no-install-recommends \
+	  autoconf \
+	  automake \
+	  build-essential \
+	  ca-certificates \
+	  ccache \
+	  cmake \
+	  cpio \
+	  file \
+	  git \
+	  libasound2-dev \
+	  libcups2-dev \
+	  libdwarf-dev \
+	  libelf-dev \
+	  libfreetype6-dev \
+	  libnuma-dev \
+	  libx11-dev \
+	  libxext-dev \
+	  libxrender-dev \
+	  libxt-dev \
+	  libxtst-dev \
+	  libssl-dev \
+	  libtool \
+	  make \
+	  nasm \
+	  pkg-config \
+	  software-properties-common \
+	  ssh \
+	  unzip \
+	  wget \
+	  zip \
+ && rm -rf /var/lib/apt/lists/*
+
+apt-get update \
+ && add-apt-repository ppa:ubuntu-toolchain-r/test -y \
+ && apt-get update \
+ && apt-get install -qq -y --no-install-recommends gcc-7 g++-7 \
+ && rm -rf /var/lib/apt/lists/*
+
+ln -sf /usr/bin/g++ /usr/bin/c++ \
+ && ln -sf /usr/bin/g++-7 /usr/bin/g++ \
+ && ln -sf /usr/bin/gcc /usr/bin/cc \
+ && ln -sf /usr/bin/gcc-7 /usr/bin/gcc
+
+cd /root \
+ && wget https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz/download -O freemarker.tar.gz \
+ && tar -xzf freemarker.tar.gz freemarker-2.3.8/lib/freemarker.jar --strip=2 \
+ && rm -f freemarker.tar.gz
+
+cd /root \
+ && wget https://api.adoptopenjdk.net/openjdk8-openj9/releases/x64_linux/latest/binary -O bootjdk8.tar.gz \
+ && tar -xzf bootjdk8.tar.gz \
+ && rm -f bootjdk8.tar.gz \
+ && mv $(ls | grep -i jdk) bootjdk8
+
+wget https://github.com/protocolbuffers/protobuf/releases/download/v3.5.1/protobuf-cpp-3.5.1.tar.gz \
```

Comment:
Do we need a specific version of `protobuf` to be installed? If there is any dependency on the version, should we point it out somewhere?

Reply:
`3.5.1` is the most installed version on X86 but `3.7.1` should be used on Z. Will update `Build.md`.